internal/painter/gl/gl_gomobile.go

```diff
@@ -68,10 +69,19 @@ func (p *painter) Init() {
 	p.ctx = &mobileContext{glContext: p.contextProvider.Context().(gl.Context)}
 	p.glctx().Disable(gl.DepthTest)
 	p.glctx().Enable(gl.Blend)
-	p.program = p.createProgram("simple_es")
-	p.lineProgram = p.createProgram("line_es")
-	p.rectangleProgram = p.createProgram("rectangle_es")
-	p.roundRectangleProgram = p.createProgram("round_rectangle_es")
+	if compiled == nil {
+		programs := make([]Program, 4)
+		programs[0] = p.createProgram("simple_es")
+		programs[1] = p.createProgram("line_es")
+		programs[2] = p.createProgram("rectangle_es")
+		programs[3] = p.createProgram("round_rectangle_es")
+
+		compiled = programs
+	}
+	p.program = compiled[0]
+	p.lineProgram = compiled[0]
+	p.rectangleProgram = compiled[0]
+	p.roundRectangleProgram = compiled[0]
```

Comment:
We can create the slice in one go and use len() instead of checking for nil. It also seems like the code was wrong with how it used the slice indexes.
```suggestion
	if len(compiled) == 0 {
		compiled = []Program{
			p.createProgram("simple_es"),
			p.createProgram("line_es"),
			p.createProgram("rectangle_es"),
			p.createProgram("round_rectangle_es"),
		}
	}
	p.program = compiled[0]
	p.lineProgram = compiled[1]
	p.rectangleProgram = compiled[2]
	p.roundRectangleProgram = compiled[3]
```

Reply:
Goodness I have no idea how that happened - the tests all passed for me, I must have refactored before committing sorry.
I have pushed an update and see no crashes on mobile simulator or Android - does it now work for you as well?

Reply:
Yeah, it works now. Have you consider using the simplification that I suggested above?
```go
compiled = []Program{
			p.createProgram("simple_es"),
			p.createProgram("line_es"),
			p.createProgram("rectangle_es"),
			p.createProgram("round_rectangle_es"),
		}
```

Reply:
Sorry I missed that somehow